### PR TITLE
Update summarise-ct-results

### PR DESCRIPTION
**Pull request by arcusfelis**
After analyzing history, it looks like the script also belongs to the repo and could be tested with tests.

### CODE REVIEW

File: test/hook_SUITE.erl [hunk 1]
```diff
@@ -68,6 +82,18 @@ run_testcases_fail_based_on_counter_repeat3(_Config) ->
...
+     "    1 groups failed",
+     "    0 tests eventually passed",
+     "    0 tests with end_per_testcase failed",
+     "    6 tests passed",
+     "    3 tests failed","    0 tests skipped by user",
+     "    0 tests skipped automatically",
+     "No test groups were executed successfully",
+     "Failing the test due to failed rerun groups",
+     "Failing the test due to failed test cases",
+     "CODE=53"] = summarise_results(),
```
Comment (chrzaszcz) [inline test/hook_SUITE.erl]: Why is the code 53 even though CT results above show lower numbers?

File: test/hook_SUITE.erl [hunk 2]
```diff
@@ -68,6 +82,18 @@ run_testcases_fail_based_on_counter_repeat3(_Config) ->
     [{groups_summary, {0, 1}},
      {eventually_ok_tests, 0},
      {end_per_testcase_failures, 0}] = GrSum,
+    ["CT results:",
+     "    0 groups passed",
+     "    1 groups failed",
+     "    0 tests eventually passed",
+     "    0 tests with end_per_testcase failed",
+     "    6 tests passed",
+     "    3 tests failed","    0 tests skipped by user",
```
Comment (chrzaszcz) [inline test/hook_SUITE.erl]: Add line break

File: test/hook_SUITE.erl [hunk 3]
```diff
@@ -84,6 +110,15 @@ run_testcases_fail_based_on_counter_repeat4(_Config) ->
     [{groups_summary, {1, 0}},
      {eventually_ok_tests, 3},
      {end_per_testcase_failures, 0}] = GrSum,
+    ["CT results:",
+     "    1 groups passed",
+     "    0 groups failed",
+     "    3 tests eventually passed",
+     "    0 tests with end_per_testcase failed",
+     "    9 tests passed",
+     "    3 tests failed","    0 tests skipped by user",
```
Comment (chrzaszcz) [inline test/hook_SUITE.erl]: Add line break

**Review by chrzaszcz — COMMENTED** [review]
It looks good in general. I added some comments. I am not completely sure how it fits into MIM - does it require any change in the MIM repo?

File: test/hook_SUITE.erl (hunk 1 above)
Comment (arcusfelis) [reply test/hook_SUITE.erl]: Oh, it is `erlang:halt(100405)`, but error code overflows.
```
1> erlang:halt(100405).
exits
echo $?
53
```

**Comment by arcusfelis** [comment]
> It looks good in general. I added some comments. I am not completely sure how it fits into MIM - does it require any change in the MIM repo?

MIM repo would have to execute `big_tests/_build/test/lib/ct_groups_summary_hook/priv/summarise-ct-results` instead of `tools/summarise-ct-results`

**Review by chrzaszcz — APPROVED** [review]
Looks good. I am not a fan of such reformatting of a whole repo, but since the history is lost anyway, we can do it.

**Pull request merged**